src/game/creature.c

```diff
@@ -913,6 +913,7 @@ int32_t __cdecl Creature_CanTargetEnemy(
     target.pos.x = enemy->pos.x;
     target.pos.y = enemy->pos.y - STEP_L * 3;
     target.pos.z = enemy->pos.z;
+    target.room_num = enemy->room_num;
```

Comment:
I'm a bit torn here; can we use the previous fix where we always set the target room number, and on top of it, have every caller set the room number as well?

This way we make both the callers and the LOS_Check API behave nicely – the latter will make sure the room number is always set to a valid value, and the former will always provide a nice input to the function.

Reply:
Another idea is to alter the LOS internals to use `XYZ_32` for `target`, and have `LOS_Check` set the `target->room_num` to `Room_FindByPos` result.

Reply:
Yes, I'm happy to put the check back into `LOS_Check`, then have each caller explicitly set it. It feels right for sure.
I think we may need to have `Gun_FindTargetPoint` decompiled as well first though, as @walkawayy mentioned:
https://github.com/LostArtefacts/TR2X/issues/137#issuecomment-2303347091

Reply:
Done

Reply:
Thanks! Looks like `Gun_FindTargetPoint` does set `target->room_num = item->room_num;`, so that situation should be good.

Reply:
Pushed a change to have all calls set the room number first, and `LOS_Check` also does a check for identical x/z. I can squash when the time comes as the title/commits have changed a bit already for this one.

[TR2X.zip](https://github.com/user-attachments/files/16717945/TR2X.zip)

The other main references to the function are in `camera.c` but everything here checks out already as far as I can tell.